query/src/lib/query-options.ts

```diff
@@ -57,6 +69,127 @@ export function queryOptions<
   queryKey: DataTag<TQueryKey, TData>;
 };
 
+export function queryOptions<
+  TQueryFnData = unknown,
+  TError = DefaultError,
+  TData = TQueryFnData,
+  TQueryKey extends QueryKey = QueryKey,
+  TPageParam = never,
+>(
+  options: CreateInfiniteQueryOptions<
+    TQueryFnData,
+    TError,
+    TData,
+    TQueryFnData,
+    TQueryKey,
+    TPageParam
+  >,
+): CreateInfiniteQueryOptions<
+  TQueryFnData,
+  TError,
+  TQueryFnData,
+  TQueryFnData,
+  TQueryKey,
+  TPageParam
+> & {
+  queryKey: DataTag<TQueryKey, TData>;
+};
 export function queryOptions(options: unknown) {
   return options;
 }
+
+export function queryObserverOptions<
+  TQueryFnData = unknown,
+  TError = DefaultError,
+  TData = TQueryFnData,
+  TQueryKey extends QueryKey = QueryKey,
+  TPageParam = never,
+>(
+  options: CreateInfiniteQueryOptions<
+    TQueryFnData,
+    TError,
+    TData,
+    TQueryFnData,
+    TQueryKey,
+    TPageParam
+  >,
+): InfiniteQueryObserverOptions<
+  TQueryFnData,
+  TError,
+  TQueryFnData,
+  TQueryFnData,
+  TQueryKey,
+  TPageParam
+>;
+export function queryObserverOptions<
+  TQueryFnData = unknown,
+  TError = DefaultError,
+  TData = TQueryFnData,
+  TQueryKey extends QueryKey = QueryKey,
+  TPageParam = never,
+>(
+  options: CreateBaseQueryOptions<
+    TQueryFnData,
+    TError,
+    TData,
+    TQueryFnData,
+    TQueryKey,
+    TPageParam
+  >,
+): QueryObserverOptions<
+  TQueryFnData,
+  TError,
+  TData,
+  TQueryFnData,
+  TQueryKey,
+  TPageParam
+>;
+export function queryObserverOptions(options: unknown) {
```

Comment:
Why we need this? I don't see any usage

Reply:
I prefer to use a distinct method to use inside queryClient overloads to convert FetchQuery/FetchInfiniteQuery to QueryObserverOptions/QueryObserverInfiniteOptions.
This way queryOptions keeps its actual use as per documentation and works with CreateInfiniteQueryOptions too otherwise this wouldn't be possibile: https://github.com/ngneat/query/blob/a1622c4330e814ea7d4e99f27c1880aae3f96959/query/src/tests/test-helper.ts#L82-L97.

Reply:
I'm not sure  I understand. You create an additional function named `queryObserverOptions`. Where we use it?

Reply:
Sorry for being so late, a lot of stuff in RL and no time to spent here. 😞 

The function **was** used in query-client where the bulk of the stuff happens:
https://github.com/ngneat/query/blob/a1622c4330e814ea7d4e99f27c1880aae3f96959/query/src/lib/query-client.ts#L126-L138

But you are right, it was useless, i'll make another commit to remove it.